src/ansys/sound/core/sound_composer/source_broadband_noise.py

```diff
@@ -0,0 +1,334 @@
+        """Return the string representation of the object."""
+        # Source info.
+        if self.source_bbn is not None:
+            spectrum_type, delta_f, control_name, control_unit, control_values = (
+                self.__extract_bbn_info()
+            )
+
+            # Source name.
+            str_name = self.source_bbn.name
+            if str_name is None:
+                str_name = ""
+
+            # Spectrum type. TODO: reactivate if/else when quantity_type is available in python.
+            # if spectrum_type == "NARROWBAND":
+            #     str_type = f"{spectrum_type} (DeltaF: {delta_f:.1f} Hz)"
+            # else:
+            #     str_type = spectrum_type
+            str_type = spectrum_type
+
+            # Spectrum control values.
+            control_values = np.round(control_values, 1)
+            if len(control_values) > 10:
+                str_values = f"{str(control_values[:5])[:-1]} ... {str(control_values[-5:])[1:]}"
+            else:
+                str_values = str(control_values)
+
+            str_source = (
+                f"'{str_name}'\n"
+                f"\tSpectrum type: {str_type}\n"
+                f"\tSpectrum count: {len(self.source_bbn)}\n"
+                f"\tControl parameter: {control_name}, {control_unit}\n"
+                f"\t\t{str_values}"
+            )
+        else:
+            str_source = "Not set"
+
+        # Source control info.
+        if self.is_source_control_valid():
+            str_source_control = (
+                f"{self.source_control.control.name}\n"
+                f"\tMin: {self.source_control.control.data.min()}\n"
+                f"\tMax: {self.source_control.control.data.max()}\n"
+                f"\tDuration: "
+                f"{self.source_control.control.time_freq_support.time_frequencies.data[-1]} s"
+            )
+        else:
+            str_source_control = "Not set"
+
+        return f"Broadband noise source: {str_source}\nSource control: {str_source_control}"
+
+    @property
+    def source_control(self) -> SourceControlTime:
+        """Broadband noise source control.
+
+        Contains the control parameter values over time.
+        """
+        return self.__source_control
+
+    @source_control.setter
+    def source_control(self, source_control: SourceControlTime):
+        """Set the source control."""
+        if not (isinstance(source_control, SourceControlTime) or source_control is None):
+            raise PyAnsysSoundException(
+                "Specified source control object must be of type ``SourceControlTime``."
+            )
+        self.__source_control = source_control
+
+    @property
+    def source_bbn(self) -> FieldsContainer:
+        """Broadband noise source data, as a DPF fields container.
+
+        The broadband noise source data consists of a series of spectra, each corresponding to a
+        control parameter value. Spectra can be narrowband, PSD, octave-band levels, or
```

Comment:
@ansaminard Spectra can be narrowband (level vs frequency or PSD), octave-band levels, or 1/3-octave-band levels

Reply:
@anshlachamb I get your point, but I feel "level vs frequency" is quite vague. I rephrased it with a reference to the unit (dB vs dB/Hz) I think it is clearer this way.